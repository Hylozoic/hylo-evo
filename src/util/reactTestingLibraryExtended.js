import React from 'react'
import { render } from '@testing-library/react'
import { MemoryRouter } from 'react-router'

const AllTheProviders = ({ children }) => {
  return (
    <MemoryRouter>
      {children}
    </MemoryRouter>
  )
}

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }
