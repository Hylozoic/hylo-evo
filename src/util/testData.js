import faker from '@faker-js/faker'
import { times } from 'lodash'

export function fakePerson (count = 1) {
  const data = times(count, () => ({
    id: faker.datatype.number(),
    name: faker.name.findName(),
    avatarUrl: faker.image.avatar() + '?' + faker.datatype.number(),
    tagline: faker.name.jobTitle()
  }))
  return count === 1 ? data[0] : data
}

export function fakePost () {
  return {
    id: 'FAKE_' + faker.datatype.number(),
    type: 'offer',
    title: 'We put this together as a PDF for hand-out at your next event or university class',
    context: 'Stop Wombat Walrus',
    details: 'Feel free to print and distribute if you would like to suggest anything we have missed or better clarity, let us know!',
    votesTotal: faker.datatype.number(),
    tags: ['activism', 'petition'],
    attachments: [{ url: 'animage', attachmentType: 'image' }],
    groups: [{ id: '1', slug: 'great', name: 'Great Cause' }],
    commenters: fakePerson(3),
    commentersTotal: faker.datatype.number(),
    creator: fakePerson(),
    updatedAt: faker.date.recent().toString()
  }
}

export const SAMPLE_IMAGE_URL = 'https://d3ngex8q79bk55.cloudfront.net/group/1944/banner/1489687099172_ggbridge.jpg'

export function fakeComment () {
  return {
    id: 'FAKE_' + faker.datatype.number(),
    creator: fakePerson(),
    createdAt: faker.date.recent().toString(),
    text: "Hey Steven! I live right next to there and can come help out. I've never done petitioning but I'm open to learn."
  }
}
