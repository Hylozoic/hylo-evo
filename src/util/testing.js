import PropTypes from 'prop-types'
import { mount } from 'enzyme'

export function mockRouter () {
  return {
    history: {
      push: jest.fn(),
      replace: jest.fn(),
      createHref: jest.fn()
    }
  }
}

export function mockRouterMountOptions () {
  return {
    context: {router: mockRouter()},
    childContextTypes: {router: PropTypes.object}
  }
}

export function mountWithMockRouter (component) {
  return mount(component, mockRouterMountOptions())
}

export const isDev = process.env.NODE_ENV === 'development'
