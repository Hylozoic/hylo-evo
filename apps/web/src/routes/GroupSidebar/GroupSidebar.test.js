import
GroupSidebar,
{
  MemberSection,
  GroupStewardsSection,
  GroupSteward
} from './GroupSidebar'
// import AboutSection from './AboutSection'
import { shallow } from 'enzyme'
import React from 'react'
import faker from '@faker-js/faker'
import { fakePerson } from 'util/testing/testData'
import { AllTheProviders, render, screen } from 'util/testing/reactTestingLibraryExtended'

const group = {
  id: 1,
  name: 'A Great Cause',
  slug: 'great-cause',
  description: 'the description, which is long enough to add a "Read More" button, ' +
    'the description, which is long enough to add a "Read More" button, ' +
    'the description, which is long enough to add a "Read More" button, ' +
    'the description, which is long enough to add a "Read More" button, ' +
    'the description, which is long enough to add a "Read More" button, '
}

const responsibilities = ['RESP_ADD_MEMBERS', 'RESP_ADMINISTRATION', 'RESP_REMOVE_MEMBERS']

describe('GroupSidebar', () => {
  const members = [{ id: 1 }, { id: 2 }, { id: 3 }]
  const stewards = [{ id: 4 }, { id: 5 }, { id: 6 }]
  const memberCount = 56
  const currentUser = { memberships: { toRefArray: () => [{ commonRoles: { items: [] } }] } }

  it('renders correctly', () => {
    const { asFragment } = render(
      <GroupSidebar
        group={{ ...group, memberCount }}
        currentUser={currentUser}
        members={members}
        stewards={stewards}
        myResponsibilities={responsibilities}
      />,
      { wrapper: AllTheProviders() }
    )
    expect(asFragment()).toMatchSnapshot()
  })
})

// describe('AboutSection', () => {
// TODO: Fix this test
//
//   it('renders correctly', () => {
//     const wrapper = shallow(
//       <AboutSection
//         name={group.name}
//         description={group.description} />)
//     expect(wrapper.find('div').at(1).text()).toEqual(`About ${group.name}`)
//     expect(wrapper.find('span').at(0).text()).toEqual('Read More')
//     expect(wrapper.find('span').at(0).text()).toEqual('Show Less')
//     wrapper.setState({ expanded: true })
//   })
// })

describe('MemberSection', () => {
  faker.seed(33)
  const n = 8
  const members = fakePerson(n)

  it("Doesn't show total if it's < 1", () => {
    const wrapper = shallow(<MemberSection
      slug='foo'
      members={members}
      memberCount={n}
      responsibilities={responsibilities}
    />)
    expect(wrapper).toMatchSnapshot()
  })

  it("Formats total correctly if it's > 999", () => {
    const wrapper = shallow(<MemberSection
      slug='foo'
      members={members}
      memberCount={5600} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('Shows invite link if has responsibility ADD_MEMBERS is true', () => {
    const wrapper = shallow(<MemberSection
      slug='foo'
      members={members}
      memberCount={5600}
      responsibilities={responsibilities} />)
    expect(wrapper).toMatchSnapshot()
  })
})

describe('GroupStewardsSection', () => {
  it('renders correctly', async () => {
    const n = 5
    const stewards = fakePerson(n)
    render(<GroupStewardsSection stewards={stewards} descriptor='Wizard' />, { wrapper: AllTheProviders() })

    expect(await screen.findByText('Group {{locationDescriptor}}')).toBeInTheDocument()
  })
})

describe('GroupSteward', () => {
  it('renders correctly', async () => {
    const steward = {
      id: 1,
      name: 'Jon Smith',
      avatarUrl: 'foo.png',
      commonRoles: { items: [] },
      groupRoles: { items: [] }
    }
    render(<GroupSteward steward={steward} />, { wrapper: AllTheProviders() })
    expect(await screen.findByText(steward.name)).toBeInTheDocument()
  })
})
