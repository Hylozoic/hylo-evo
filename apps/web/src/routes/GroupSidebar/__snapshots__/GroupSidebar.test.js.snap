// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupSidebar renders correctly 1`] = `
<DocumentFragment>
  <div
    data-stylename="group-sidebar"
  >
    <div
      data-stylename="about-section"
    >
      <div
        data-stylename="about-content"
      >
        <span>
          <h2>
            Group Description
          </h2>
          <span>
            <p>
              the description, which is long enough to add a "Read More" button, the description, which is long enough to add a "Read More" button, the description, which is long enough to add a "Read More" button, the description, which is long enough to add a "Read More" button, the description, which is long enough to add a "Read More" button, 
            </p>
            

          </span>
        </span>
      </div>
    </div>
    <div
      data-stylename="member-section"
    >
      <a
        data-stylename="members-link"
        href="/groups/great-cause/members"
      >
        <div
          data-stylename="header"
        >
          Members
        </div>
        <div
          data-stylename="images-and-count"
        >
          <div
            data-stylename="image-row"
          >
            <div
              data-stylename="image overlaps medium"
            />
            <div
              data-stylename="image overlaps medium"
            />
            <div
              data-stylename="image overlaps medium"
            />
          </div>
          <span
            data-stylename="members-total"
          >
            +53
          </span>
        </div>
      </a>
    </div>
    <div
      data-stylename="leader-section"
    >
      <div
        data-stylename="header leader-header"
      >
        Group {{locationDescriptor}}
      </div>
      <div
        data-stylename="leader"
      >
        <a
          href="/groups/great-cause/members/4"
        >
          <div
            data-stylename="image medium"
          />
        </a>
        <a
          data-stylename="leader-name"
          href="/groups/great-cause/members/4"
        />
        <div
          data-stylename="badges"
        />
      </div>
      <div
        data-stylename="leader"
      >
        <a
          href="/groups/great-cause/members/5"
        >
          <div
            data-stylename="image medium"
          />
        </a>
        <a
          data-stylename="leader-name"
          href="/groups/great-cause/members/5"
        />
        <div
          data-stylename="badges"
        />
      </div>
      <div
        data-stylename="leader"
      >
        <a
          href="/groups/great-cause/members/6"
        >
          <div
            data-stylename="image medium"
          />
        </a>
        <a
          data-stylename="leader-name"
          href="/groups/great-cause/members/6"
        />
        <div
          data-stylename="badges"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`MemberSection Doesn't show total if it's < 1 1`] = `
<div
  data-stylename="member-section"
>
  <Link
    data-stylename="members-link"
    to="/groups/foo/members"
  >
    <div
      data-stylename="header"
    >
      Members
    </div>
    <div
      data-stylename="images-and-count"
    >
      <RoundImageRow
        data-stylename="image-row"
        imageUrls={
          Array [
            "https://cloudflare-ipfs.com/ipfs/Qmd3W5DuhgHirLHGVixi6V76LhCkZUz6pnFt5AJBiyvHye/avatar/1150.jpg?26029",
            "https://cloudflare-ipfs.com/ipfs/Qmd3W5DuhgHirLHGVixi6V76LhCkZUz6pnFt5AJBiyvHye/avatar/1099.jpg?68045",
            "https://cloudflare-ipfs.com/ipfs/Qmd3W5DuhgHirLHGVixi6V76LhCkZUz6pnFt5AJBiyvHye/avatar/439.jpg?10715",
            "https://cloudflare-ipfs.com/ipfs/Qmd3W5DuhgHirLHGVixi6V76LhCkZUz6pnFt5AJBiyvHye/avatar/501.jpg?36352",
            "https://cloudflare-ipfs.com/ipfs/Qmd3W5DuhgHirLHGVixi6V76LhCkZUz6pnFt5AJBiyvHye/avatar/178.jpg?42174",
            "https://cloudflare-ipfs.com/ipfs/Qmd3W5DuhgHirLHGVixi6V76LhCkZUz6pnFt5AJBiyvHye/avatar/749.jpg?66249",
            "https://cloudflare-ipfs.com/ipfs/Qmd3W5DuhgHirLHGVixi6V76LhCkZUz6pnFt5AJBiyvHye/avatar/414.jpg?52422",
            "https://cloudflare-ipfs.com/ipfs/Qmd3W5DuhgHirLHGVixi6V76LhCkZUz6pnFt5AJBiyvHye/avatar/369.jpg?90195",
          ]
        }
      />
    </div>
  </Link>
</div>
`;

exports[`MemberSection Formats total correctly if it's > 999 1`] = `
<div
  data-stylename="member-section"
>
  <Link
    data-stylename="members-link"
    to="/groups/foo/members"
  >
    <div
      data-stylename="header"
    >
      Members
    </div>
    <div
      data-stylename="images-and-count"
    >
      <RoundImageRow
        data-stylename="image-row"
        imageUrls={
          Array [
            "https://cloudflare-ipfs.com/ipfs/Qmd3W5DuhgHirLHGVixi6V76LhCkZUz6pnFt5AJBiyvHye/avatar/1150.jpg?26029",
            "https://cloudflare-ipfs.com/ipfs/Qmd3W5DuhgHirLHGVixi6V76LhCkZUz6pnFt5AJBiyvHye/avatar/1099.jpg?68045",
            "https://cloudflare-ipfs.com/ipfs/Qmd3W5DuhgHirLHGVixi6V76LhCkZUz6pnFt5AJBiyvHye/avatar/439.jpg?10715",
            "https://cloudflare-ipfs.com/ipfs/Qmd3W5DuhgHirLHGVixi6V76LhCkZUz6pnFt5AJBiyvHye/avatar/501.jpg?36352",
            "https://cloudflare-ipfs.com/ipfs/Qmd3W5DuhgHirLHGVixi6V76LhCkZUz6pnFt5AJBiyvHye/avatar/178.jpg?42174",
            "https://cloudflare-ipfs.com/ipfs/Qmd3W5DuhgHirLHGVixi6V76LhCkZUz6pnFt5AJBiyvHye/avatar/749.jpg?66249",
            "https://cloudflare-ipfs.com/ipfs/Qmd3W5DuhgHirLHGVixi6V76LhCkZUz6pnFt5AJBiyvHye/avatar/414.jpg?52422",
            "https://cloudflare-ipfs.com/ipfs/Qmd3W5DuhgHirLHGVixi6V76LhCkZUz6pnFt5AJBiyvHye/avatar/369.jpg?90195",
          ]
        }
      />
      <span
        data-stylename="members-total"
      >
        +5.6k
      </span>
    </div>
  </Link>
</div>
`;

exports[`MemberSection Shows invite link if has responsibility ADD_MEMBERS is true 1`] = `
<div
  data-stylename="member-section"
>
  <Link
    data-stylename="members-link"
    to="/groups/foo/members"
  >
    <div
      data-stylename="header"
    >
      Members
    </div>
    <div
      data-stylename="images-and-count"
    >
      <RoundImageRow
        data-stylename="image-row"
        imageUrls={
          Array [
            "https://cloudflare-ipfs.com/ipfs/Qmd3W5DuhgHirLHGVixi6V76LhCkZUz6pnFt5AJBiyvHye/avatar/1150.jpg?26029",
            "https://cloudflare-ipfs.com/ipfs/Qmd3W5DuhgHirLHGVixi6V76LhCkZUz6pnFt5AJBiyvHye/avatar/1099.jpg?68045",
            "https://cloudflare-ipfs.com/ipfs/Qmd3W5DuhgHirLHGVixi6V76LhCkZUz6pnFt5AJBiyvHye/avatar/439.jpg?10715",
            "https://cloudflare-ipfs.com/ipfs/Qmd3W5DuhgHirLHGVixi6V76LhCkZUz6pnFt5AJBiyvHye/avatar/501.jpg?36352",
            "https://cloudflare-ipfs.com/ipfs/Qmd3W5DuhgHirLHGVixi6V76LhCkZUz6pnFt5AJBiyvHye/avatar/178.jpg?42174",
            "https://cloudflare-ipfs.com/ipfs/Qmd3W5DuhgHirLHGVixi6V76LhCkZUz6pnFt5AJBiyvHye/avatar/749.jpg?66249",
            "https://cloudflare-ipfs.com/ipfs/Qmd3W5DuhgHirLHGVixi6V76LhCkZUz6pnFt5AJBiyvHye/avatar/414.jpg?52422",
            "https://cloudflare-ipfs.com/ipfs/Qmd3W5DuhgHirLHGVixi6V76LhCkZUz6pnFt5AJBiyvHye/avatar/369.jpg?90195",
          ]
        }
      />
      <span
        data-stylename="members-total"
      >
        +5.6k
      </span>
    </div>
  </Link>
</div>
`;

exports[`MemberSection Shows invite link if has responsibility ADD_MEMBERS is true 1`] = `
<div
  data-stylename="member-section"
>
  <Link
    data-stylename="members-link"
    to="/groups/foo/members"
  >
    <div
      data-stylename="header"
    >
      Members
    </div>
    <div
      data-stylename="images-and-count"
    >
      <RoundImageRow
        data-stylename="image-row"
        imageUrls={
          Array [
            "https://cloudflare-ipfs.com/ipfs/Qmd3W5DuhgHirLHGVixi6V76LhCkZUz6pnFt5AJBiyvHye/avatar/1150.jpg?26029",
            "https://cloudflare-ipfs.com/ipfs/Qmd3W5DuhgHirLHGVixi6V76LhCkZUz6pnFt5AJBiyvHye/avatar/1099.jpg?68045",
            "https://cloudflare-ipfs.com/ipfs/Qmd3W5DuhgHirLHGVixi6V76LhCkZUz6pnFt5AJBiyvHye/avatar/439.jpg?10715",
            "https://cloudflare-ipfs.com/ipfs/Qmd3W5DuhgHirLHGVixi6V76LhCkZUz6pnFt5AJBiyvHye/avatar/501.jpg?36352",
            "https://cloudflare-ipfs.com/ipfs/Qmd3W5DuhgHirLHGVixi6V76LhCkZUz6pnFt5AJBiyvHye/avatar/178.jpg?42174",
            "https://cloudflare-ipfs.com/ipfs/Qmd3W5DuhgHirLHGVixi6V76LhCkZUz6pnFt5AJBiyvHye/avatar/749.jpg?66249",
            "https://cloudflare-ipfs.com/ipfs/Qmd3W5DuhgHirLHGVixi6V76LhCkZUz6pnFt5AJBiyvHye/avatar/414.jpg?52422",
            "https://cloudflare-ipfs.com/ipfs/Qmd3W5DuhgHirLHGVixi6V76LhCkZUz6pnFt5AJBiyvHye/avatar/369.jpg?90195",
          ]
        }
      />
      <span
        data-stylename="members-total"
      >
        +5.6k
      </span>
    </div>
  </Link>
</div>
`;
